raiden/tests/integration/long_running/test_settlement.py

```diff
@@ -880,18 +878,46 @@ def run_test_batch_unlock_after_restart(raiden_network, token_addresses, deposit
         partner_address=alice_app.raiden.address,
     )
 
-    alice_app.stop()
+    # wait for the close transaction to be mined, this is necessary to compute
+    # the timeout for the settle
+    with gevent.Timeout(timeout):
```

Comment:
wrapped all blocking calls with a timeout, otherwise the test will run until pytest-timeout raises, which takes much longer to fail.